### ISSUE TITLE
BigQuery: Document the use of timeout parameter in samples

### ISSUE THREAD

**Issue by plamut** [googleapis/google-cloud-python#9988]
After adding the new `timeout` parameter to various public methods (#9987), we should demonstrate its usage in the code samples.

Users should be aware of this new feature, and should probably use it by default to avoid sporadic weird issues related to a method "getting stuck" at the transport layer.

**Comment by plamut** [comment]
We don't have to complicate, simply adding an explicit `timeout` argument to various `client.method()` should be enough.

@tswast What do you think?

**Comment by tswast** [comment]
Makes sense to me. I think that once we have a `timeout` parameter on all relevant methods, we can surface an example in an API best practices doc such as https://cloud.google.com/bigquery/docs/api-performance

Since usage should be common across all methods, a single timeout example should be sufficient, perhaps on `create_dataset`, since that method is pretty straightforward.

Internally at Google, setting a timeout is pretty much always considered a best practice, so I wouldn't be all that opposed to adding a reasonable-length timeout to most/all samples someday, but we'd want to be careful to pick timeouts that won't make the tests super flakey.
